FHEM/14_SD_WS.pm

```diff
@@ -789,9 +816,12 @@ sub SD_WS_Parse($$)
 		}
 		Log3 $iohash,4, "$name: using longid for $longids device $deviceCode";
 	} else {
-		$deviceCode = $model . "_" . $channel;
+		if (defined $channel) {
+			$deviceCode = $model . "_" . $channel;
+		} else {
+			$deviceCode = $model . "_" . $id;	# for sensors without channel
```

Comment:
Das Kollidiert ein bisschen mit der longids Option. Nur wenn die aktiviert ist, soll ja die RandomID in der Definition verwendet werden. Bei Geräten die keinen Kanal besitzen, würde dann nur eine Definition existieren.

Reply:
Wie soll ich dann Sensoren unterscheiden, die zwar verschiedene IDs verwenden, aber keinen Kanal?
Ich hatte auch zuerst überlegt, $channel=$id zu setzen, aber so scheint mir die Lösung universeller.

Reply:
Zu longids ist folgendes dokumentiert:

> Comma separated list of device-types for SIGNALduino that should be handled using long IDs. This additional ID allows it to differentiate some weather sensors, if they are sending on the same channel. Therfor a random generated id is added. If you choose to use longids, then you'll have to define a different device after battery change.
Default is to not to use long IDs for all devices. 


Ich würde behaupten es ist wie bei allen Sensoren mit Kanal. Möchte man mehr Sensoren verwenden als es Kanäle gibt, so muss man das Attribut longids setzen.
In diesem Fall wäre das halt bereit bei mehr als einem Sensor notwendig.

Reply:
Dann vieleicht so:
	} else {
		$deviceCode = $model;	# for sensors without channel
		$deviceCode .= '_' . $channel if (defined $channel);
	}

Der Sensor wird dann so angelegt: SD_WS_94_T
und mit longids: SD_WS_94_T_0C
Gefällt mir eigentlich auch besser, warm nicht gleich so :-)